packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/RestClientProvider.cs

```diff
@@ -180,6 +163,46 @@ private IReadOnlyList<MethodBodyStatement> GetSetContent(HttpRequestApi request,
             return contentParam is null ? [] : [request.Content().Assign(contentParam).Terminate()];
         }
 
+        private Dictionary<int, PropertyProvider> BuildPipelineMessage20xClassifiers()
+        {
+            Dictionary<int, PropertyProvider> classifiers = [];
+
+            foreach (var inputOperation in _inputClient.Operations)
+            {
+                foreach (var response in inputOperation.Responses)
+                {
+                    if (response.IsErrorResponse || response.StatusCodes.Count != 1)
```

Comment:
What does it mean if there is more than one status code?

Reply:
ah given the existing logic, I assumed that having more than one success status code wasn't valid, or was not good practice. I guess a user can define an op like so:
```
op read(): {
  @statusCode _: 201 | 202;
};
```
and we should generate a classifier that classifies both statuses as successful. I'll update the pr accordingly..